packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -367,276 +471,278 @@ private void createCaptureSessionWithSessionConfig(
-        (code, message) -> pictureCaptureRequest.error(code, message, null));
-  }
+  /**
+   * Capture a still picture. This method should be called when we get a response in {@link
+   * #cameraCaptureCallback} from both lockFocus().
+   */
+  private void takePictureAfterPrecapture() {
+    Log.i(TAG, "captureStillPicture");
+    cameraCaptureCallback.setCameraState(CameraState.STATE_CAPTURING);
 
-  private void runPictureCapture() {
-    assert (pictureCaptureRequest != null);
-    pictureCaptureRequest.setState(PictureCaptureRequest.State.capturing);
     try {
-      final CaptureRequest.Builder captureBuilder =
+      if (null == cameraDevice) {
+        return;
+      }
+      // This is the CaptureRequest.Builder that we use to take a picture.
+      final CaptureRequest.Builder stillBuilder =
           cameraDevice.createCaptureRequest(CameraDevice.TEMPLATE_STILL_CAPTURE);
-      captureBuilder.addTarget(pictureImageReader.getSurface());
-      captureBuilder.set(
+      stillBuilder.addTarget(pictureImageReader.getSurface());
+
+      // Zoom
+      stillBuilder.set(
           CaptureRequest.SCALER_CROP_REGION,
-          captureRequestBuilder.get(CaptureRequest.SCALER_CROP_REGION));
-      captureBuilder.set(
+          previewRequestBuilder.get(CaptureRequest.SCALER_CROP_REGION));
+
+      // Update builder settings
+      updateBuilderSettings(stillBuilder);
+
+      // Orientation
+      final PlatformChannel.DeviceOrientation lockedOrientation =
+          ((SensorOrientationFeature) cameraFeatures.getSensorOrientation())
+              .getLockedCaptureOrientation();
+      stillBuilder.set(
           CaptureRequest.JPEG_ORIENTATION,
-          lockedCaptureOrientation == null
-              ? deviceOrientationListener.getMediaOrientation()
-              : deviceOrientationListener.getMediaOrientation(lockedCaptureOrientation));
-
-      switch (flashMode) {
-        case off:
-          captureBuilder.set(CaptureRequest.CONTROL_AE_MODE, CaptureRequest.CONTROL_AE_MODE_ON);
-          captureBuilder.set(CaptureRequest.FLASH_MODE, CaptureRequest.FLASH_MODE_OFF);
-          break;
-        case auto:
-          captureBuilder.set(
-              CaptureRequest.CONTROL_AE_MODE, CaptureRequest.CONTROL_AE_MODE_ON_AUTO_FLASH);
-          break;
-        case always:
-        default:
-          captureBuilder.set(
-              CaptureRequest.CONTROL_AE_MODE, CaptureRequest.CONTROL_AE_MODE_ON_ALWAYS_FLASH);
-          break;
-      }
-      cameraCaptureSession.stopRepeating();
-      cameraCaptureSession.capture(
-          captureBuilder.build(),
+          lockedOrientation == null
+              ? getDeviceOrientationManager().getPhotoOrientation()
+              : getDeviceOrientationManager().getPhotoOrientation(lockedOrientation));
+
+      CameraCaptureSession.CaptureCallback captureCallback =
           new CameraCaptureSession.CaptureCallback() {
+
+            @Override
+            public void onCaptureStarted(
```

Comment:
does it really need to log or can this be method with default implementation, so no need to override?

https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html